presto-tests/src/test/java/io/prestosql/execution/TestCoordinatorDynamicFiltering.java

```diff
@@ -0,0 +1,297 @@
+
+@Test(singleThreaded = true)
+public class TestCoordinatorDynamicFiltering
+        extends AbstractTestQueryFramework
+{
+    private final AtomicReference<TupleDomain<ColumnHandle>> expectedDynamicFilter = new AtomicReference<>(TupleDomain.all());
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("test")
+                .setSchema("default")
+                .setSystemProperty(JOIN_REORDERING_STRATEGY, FeaturesConfig.JoinReorderingStrategy.NONE.name())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, FeaturesConfig.JoinDistributionType.PARTITIONED.name())
+                .build();
+        return DistributedQueryRunner.builder(session).build();
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        getQueryRunner().installPlugin(new TestPlugin(expectedDynamicFilter));
+        getQueryRunner().installPlugin(new TpchPlugin());
+        getQueryRunner().createCatalog("test", "test", ImmutableMap.of());
+        getQueryRunner().createCatalog("tpch", "tpch", ImmutableMap.of());
+        computeActual("CREATE TABLE lineitem AS SELECT * FROM tpch.tiny.lineitem");
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void afterMethod()
+    {
+        expectedDynamicFilter.set(TupleDomain.all());
+    }
+
+    @Test
+    public void testWithEmptyBuildSide()
+    {
+        assertQueryDynamicFilters(
+                "SELECT * FROM lineitem JOIN tpch.tiny.supplier ON lineitem.suppkey = supplier.suppkey AND supplier.name = 'abc'",
+                TupleDomain.none());
+    }
+
+    @Test
+    public void testWithSelectiveBuildSide()
+    {
+        assertQueryDynamicFilters(
+                "SELECT * FROM lineitem JOIN tpch.tiny.supplier ON lineitem.suppkey = supplier.suppkey AND supplier.name = 'Supplier#000000001'",
+                TupleDomain.withColumnDomains(ImmutableMap.of(
+                        new TestingMetadata.TestingColumnHandle("suppkey", 2, BIGINT),
+                        Domain.singleValue(BIGINT, 1L))));
+    }
+
+    @Test
+    public void testWithNonSelectiveBuildSide()
+    {
+        assertQueryDynamicFilters(
+                "SELECT * FROM lineitem JOIN tpch.tiny.supplier ON lineitem.suppkey = supplier.suppkey",
+                TupleDomain.withColumnDomains(ImmutableMap.of(
+                        new TestingMetadata.TestingColumnHandle("suppkey", 2, BIGINT),
+                        Domain.create(ValueSet.ofRanges(Range.range(BIGINT, 1L, true, 100L, true)), false))));
+    }
+
+    @Test
+    public void testWithMultipleDynamicFiltersOnProbe()
+    {
+        assertQueryDynamicFilters(
+                "SELECT * FROM (" +
+                        "SELECT supplier.suppkey FROM " +
+                            "lineitem JOIN tpch.tiny.supplier ON lineitem.suppkey = supplier.suppkey AND supplier.name IN ('Supplier#000000001', 'Supplier#000000002')" +
```

Comment:
add a comment what key values `'Supplier#000000001', 'Supplier#000000002'` match to.